### ISSUE TITLE
关于convnext_end.pth和replk_end.pth是如何获得，以及如何对测试集进行测试

### ISSUE THREAD

**Issue by allmight1110** [VisionRush/DeepFakeDefenders#23]
运行main_infer.py会报错
![1726400386603](https://github.com/user-attachments/assets/2ed8656c-7014-4583-8522-48c53a29e802)
请问怎么解决？

**Comment by Fieldhunter** [comment]
您好，如果可以的话，请把所有的报错信息都列出来看一下

**Comment by allmight1110** [comment]
> 您好，如果可以的话，请把所有的报错信息都列出来看一下

![7c0c7f8dad7e34ec06d1962b52a286a](https://github.com/user-attachments/assets/6b55ca35-46ec-4f17-95f4-86734e907aa0)
您好 这里是完整的报错信息。

**Comment by Fieldhunter** [comment]
很明显，这是由于您的模型权重文件与代码中定义的模型结构不匹配造成的，看报错信息似乎您的模型权重文件是没有错误的。请确认在main_infer.py中定义模型时，是否加载了toolkit/chelper.py中的final_model模型（即load_model函数的model_name参数传入'all'）。

**Comment by allmight1110** [comment]
> 很明显，这是由于您的模型权重文件与代码中定义的模型结构不匹配造成的，看报错信息似乎您的模型权重文件是没有错误的。请确认在main_infer.py中定义模型时，是否加载了toolkit/chelper.py中的final_model模型（即load_model函数的model_name参数传入'all'）。
您好，首先感谢您的开源代码和解答 这部分我们已经解决了，但是现在遇到了一个新问题 。在复现的过程中，在推理的这一步报错，关于main_infer里的测试 img = '51aa9b8d0da890cd1d0c5029e3d89e3c.jpg'经过处理后形状为[1,6,6,512,512]的维度问题
def main():
img = '51aa9b8d0da890cd1d0c5029e3d89e3c.jpg'
infer_api = INFER_API()
print(infer_api.test(img))
if name == 'main':
main()
这里图片的embedding为[1,6,6,512,512]
但是RuntimeError: Expected 3D (unbatched) or 4D (batched) input to conv2d, but got input of size: [1, 6, 6, 512, 512]
![777cd166f66f946c28954a48810fe4d](https://github.com/user-attachments/assets/a5a93bd6-cbe0-4702-8cf0-827d2e076c84)

我们尝试这打印每一步变换的图片形状
![1f2a08b17aa98ae786f24c6c2df4903](https://github.com/user-attachments/assets/8087335c-9fc8-43a1-8fcb-0bb99de5e88d)
![b233f543604623c0dc3226178247826](https://github.com/user-attachments/assets/b4407626-50b5-4201-8ebc-99e9f53354e7)请问输入到模型中的图片维度要求是什么，这里测试一张图片但是经过六种变换之后到all_data列表里面是6个[6,512,512]的张量，经过 img_tensor = torch.stack(all_data, dim=0).unsqueeze(0).cuda()之后变成了[1,6,6,512,512,]导致输入报错输入形状错误

**Comment by Fieldhunter** [comment]
> > 很明显，这是由于您的模型权重文件与代码中定义的模型结构不匹配造成的，看报错信息似乎您的模型权重文件是没有错误的。请确认在main_infer.py中定义模型时，是否加载了toolkit/chelper.py中的final_model模型（即load_model函数的model_name参数传入'all'）。
> > 您好，首先感谢您的开源代码和解答 这部分我们已经解决了，但是现在遇到了一个新问题 。在复现的过程中，在推理的这一步报错，关于main_infer里的测试 img = '51aa9b8d0da890cd1d0c5029e3d89e3c.jpg'经过处理后形状为[1,6,6,512,512]的维度问题
> > def main():
> > img = '51aa9b8d0da890cd1d0c5029e3d89e3c.jpg'
> > infer_api = INFER_API()
> > print(infer_api.test(img))
> > if name == 'main':
> > main()
> > 这里图片的embedding为[1,6,6,512,512]
> > 但是RuntimeError: Expected 3D (unbatched) or 4D (batched) input to conv2d, but got input of size: [1, 6, 6, 512, 512]
> > ![777cd166f66f946c28954a48810fe4d](https://private-user-images.githubusercontent.com/67216495/368956326-a5a93bd6-cbe0-4702-8cf0-827d2e076c84.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3NDM4MDMsIm5iZiI6MTcyNjc0MzUwMywicGF0aCI6Ii82NzIxNjQ5NS8zNjg5NTYzMjYtYTVhOTNiZDYtY2JlMC00NzAyLThjZjAtODI3ZDJlMDc2Yzg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE5VDEwNTgyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5ZDAzNjdmNGNjNGQ5YWIxMjQzM2U1YTg4N2Y2OTY0NDIxM2RmZjg1Y2QwODIzMWNlYzQ4ZmVjOGRiMDlhMTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.otARYT2Hk0aNziE4pqtC2OHCcpgPiVER6eq4Bu9-0nw)
> 
> 我们尝试这打印每一步变换的图片形状 ![1f2a08b17aa98ae786f24c6c2df4903](https://private-user-images.githubusercontent.com/67216495/368956380-8087335c-9fc8-43a1-8fcb-0bb99de5e88d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3NDM4MDMsIm5iZiI6MTcyNjc0MzUwMywicGF0aCI6Ii82NzIxNjQ5NS8zNjg5NTYzODAtODA4NzMzNWMtOWZjOC00M2ExLThmY2ItMGJiOTlkZTVlODhkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE5VDEwNTgyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2ZGRmYzhjY2NhMzQ0ZDU4MDkwOGU0MzRkZmRlODVjNmVlNzVmM2ZiYjM3N2VjNDdiYzhkNzUzMDllNTM5NGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.f5RQ8eL_4sE2CQawUYMKaAJ6WV1KciWSOt1PjfUbm6w) ![b233f543604623c0dc3226178247826](https://private-user-images.githubusercontent.com/67216495/368956407-b4407626-50b5-4201-8ebc-99e9f53354e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3NDM4MDMsIm5iZiI6MTcyNjc0MzUwMywicGF0aCI6Ii82NzIxNjQ5NS8zNjg5NTY0MDctYjQ0MDc2MjYtNTBiNS00MjAxLThlYmMtOTllOWY1MzM1NGU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE5VDEwNTgyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4ZGMyZWViYTA1Yjg3ZTBiMWI0MTUxNTVjMTc0NDAyMjQyOWJiODVhMTZhNmJlNTJkMzZiODM1ZjUwYmIyN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7lQXBEgEOZGqepLnLIBv3MJcSgJYGyhukn3UMk0vxSY)请问输入到模型中的图片维度要求是什么，这里测试一张图片但是经过六种变换之后到all_data列表里面是6个[6,512,512]的张量，经过 img_tensor = torch.stack(all_data, dim=0).unsqueeze(0).cuda()之后变成了[1,6,6,512,512,]导致输入报错输入形状错误

您好，这个问题请移步 #25
